ui/raidboss/data/03-hw/alliance/weeping_city.js

```diff
@@ -0,0 +1,312 @@
+      netRegex: NetRegexes.headMarker({ id: '0039' }),
+      condition: Conditions.targetIsYou(),
+      alertText: {
+        en: 'Drop meteor back or left',
+      },
+    },
+    {
+      // The ability used here is Ozma entering Pyramid form.
+      // Execration follows this up almost immediately.
+      id: 'Weeping City Execration',
+      netRegex: NetRegexes.ability({ id: '1826', source: 'Ozma', capture: false }),
+      alertText: {
+        en: 'Get off rectangle platform',
+      },
+    },
+    {
+      // The ability used here is Ozma entering Cube form.
+      // Flare Star and Explosion follow shortly.
+      id: 'Weeping City Flare Star',
+      netRegex: NetRegexes.ability({ id: '1803', source: 'Ozma', capture: false }),
+      alertText: function(data) {
+        if (data.role == 'tank') {
+          return {
+            en: 'Tank lasers--Avoid party',
+          };
+        }
+        if (data.role == 'healer') {
+          return {
+            en: 'Avoid tanks, get orbs',
+          };
+        }
+        return {
+          en: 'Avoid tanks and healers',
+        };
+      },
+    },
+    {
+      id: 'Weeping City Acceleration Bomb',
+      netRegex: NetRegexes.gainsEffect({ effectId: '430' }),
+      condition: Conditions.targetIsYou(),
+      delaySeconds: function(data, matches) {
+        return parseFloat(matches.duration) - 3;
+      },
+      response: Responses.stopEverything(),
+    },
+    {
+      id: 'Weeping City Assimilation',
+      netRegex: NetRegexes.startsUsing({ id: '1802', source: 'Ozmashade', capture: false }),
+      response: Responses.lookAway(),
+    },
+    {
+      // Each party gets a stack marker, so this is the best we can do.
+      id: 'Weeping City Meteor Stack',
+      netRegex: NetRegexes.headMarker({ id: '003E', capture: false }),
+      condition: function(data) {
+        return data.ozmaStarted;
+      },
+      suppressSeconds: 5,
+      response: Responses.stack(),
+    },
+    {
+      // Coif Change is always followed up shortly by Haircut.
+      // There's no castbar or indicator except that she grows a scythe on one side.
+      // It's not a very obvious visual cue unless the player knows to look for it.
+      id: 'Weeping City Coif Change Left',
+      netRegex: NetRegexes.ability({ id: '180A', source: 'Calofisteri', capture: false }),
+      response: Responses.goRight(),
+    },
+    {
+      id: 'Weeping City Coif Change Right',
+      netRegex: NetRegexes.ability({ id: '180E', source: 'Calofisteri', capture: false }),
+      response: Responses.goLeft(),
+    },
+    {
+      // 4899 is the base ID for bulb hair. 4900 is axe hair.
+      // Bulbs do a circle AoE surrounding them, while axes are a donut.
+      id: 'Weeping City Living Lock Axes',
+      netRegex: NetRegexes.addedCombatantFull({ npcNameId: ['4899', '4900'], capture: false }),
+      suppressSeconds: 5,
+      infoText: {
+        en: 'Close to axes, far from bulbs',
```

Comment:
"Get Under Axe", for brevity?

Reply:
There's at least one pattern with no axes, unfortunately. Special-case that?

Reply:
I have never seen no axes before! I guess this is fine then to avoid complicated logic.